### DISCUSSION TITLE
How do I move a project to another cleint

### DISCUSSION THREAD

**Discussion by zyll71** [te-online/timemanager#191] [Q&A]
Regulations have changed and I need to retrospectively re-assign projects and tasks to other internal clients for this year. The project's edit dialog does not allow me to make that change. How do I move a project to a different client?

![image](https://github.com/user-attachments/assets/38ba0fdc-d86a-469e-85ea-611c9822bbb4)

**Comment by te-online** [comment]
I'm sorry I don't have good news for you – it's not possible to change a project's associated client through the UI.

However, you can re-assign the project(s) in the database, if this is very important to you.

> [!CAUTION]
> **REMEMBER to make a backup of your database before making ANY changes**

Here's a step-by-step guide:

1. We want to move "Testproject" from "Testclient" to "Testclient 2"
![grafik](https://github.com/user-attachments/assets/8fadb024-2ebe-4653-8db1-686ee81d5867)
![Screenshot 2024-10-31 at 17 25 13](https://github.com/user-attachments/assets/216c18a3-6645-4f90-a5d6-ae3cfb5c6ee5)

2. Connect to the database and open the table `oc_timemanager_project`, your prefix may be something other than `oc_`
![Screenshot 2024-10-31 at 17 26 40](https://github.com/user-attachments/assets/ed57e4d4-6dbb-4d76-9e7f-afd3aaecaab2)

3. Identify the projects you want to move using their title or uuid (you can see the uuid in the URL when you're on the project page)
4. Change the `client_uuid` to the `uuid` that matches the "Testclient 2" (taken from the URL in Nextcloud or the clients table in the database)
![Screenshot 2024-10-31 at 17 28 45](https://github.com/user-attachments/assets/8c8e47e2-fbb7-4c34-b0ee-cbdf67d29bc3)

5. Double-check the project(s) have moved
![grafik](https://github.com/user-attachments/assets/b503bc88-468a-4ed2-a638-1818a0d57f71)

**Comment by zyll71** [comment]
I'm neither shocked nor intimidated by your answer. Only grateful for your workaround instructions. I'll give it a go. Thanks!

On 31 October 2024 17:35:03 CET, Thomas Ebert ***@***.***> wrote:
>I'm sorry I don't have good news for you – it's not possible to change a project's associated client through the UI.
>
>However, you can re-assign the project(s) in the database, if this is very important to you.
>
>> [!CAUTION]
>> **REMEMBER to make a backup of your database before making ANY changes**
>
>Here's a step-by-step guide:
>
>1. We want to move "Testproject" from "Testclient" to "Testclient 2"
>![grafik](https://github.com/user-attachments/assets/8fadb024-2ebe-4653-8db1-686ee81d5867)
>![Screenshot 2024-10-31 at 17 25 13](https://github.com/user-attachments/assets/216c18a3-6645-4f90-a5d6-ae3cfb5c6ee5)
>
>2. Connect to the database and open the table `oc_timemanager_project`, your prefix may be something other than `oc_`
>![Screenshot 2024-10-31 at 17 26 40](https://github.com/user-attachments/assets/ed57e4d4-6dbb-4d76-9e7f-afd3aaecaab2)
>
>3. Identify the projects you want to move using their title or uuid (you can see the uuid in the URL when you're on the project page)
>4. Change the `client_uuid` to the `uuid` that matches the "Testclient 2" (taken from the URL in Nextcloud or the clients table in the database)
>![Screenshot 2024-10-31 at 17 28 45](https://github.com/user-attachments/assets/8c8e47e2-fbb7-4c34-b0ee-cbdf67d29bc3)
>
>5. Double-check the project(s) have moved
>![grafik](https://github.com/user-attachments/assets/b503bc88-468a-4ed2-a638-1818a0d57f71)
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/te-online/timemanager/discussions/191#discussioncomment-11114031
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
